microsetta_admin/metadata_util.py

```diff
@@ -1,141 +1,258 @@
 from microsetta_admin._api import APIRequest
-from collections import defaultdict
+from microsetta_admin.metadata_constants import HUMAN_SITE_INVARIANTS
+from collections import Counter
+import re
+import pandas as pd
+
+
+EBI_REMOVE = ['ABOUT_YOURSELF_TEXT', 'ANTIBIOTIC_CONDITION',
+              'ANTIBIOTIC_MED',
+              'BIRTH_MONTH', 'CAT_CONTACT', 'CAT_LOCATION',
+              'CONDITIONS_MEDICATION', 'DIET_RESTRICTIONS_LIST',
+              'DOG_CONTACT',
+              'DOG_LOCATION', 'GENDER', 'MEDICATION_LIST',
+              'OTHER_CONDITIONS_LIST', 'PREGNANT_DUE_DATE',
+              'RACE_OTHER',
+              'RELATIONSHIPS_WITH_OTHERS_IN_STUDY',
+              'SPECIAL_RESTRICTIONS',
+              'SUPPLEMENTS', 'TRAVEL_LOCATIONS_LIST', 'ZIP_CODE',
+              'WILLING_TO_BE_CONTACTED', 'pets_other_freetext']
+
+
+def drop_private_columns(df):
+    """Remove columns that should not be shared publicly
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The dataframe to operate on
+
+    Returns
+    -------
+    pd.DataFrame
+        The filtered dataframe
+    """
+    # The personal microbiome survey contains additional fields that are
+    # sensitive in nature
+    pm_remove = {c.lower() for c in df.columns if c.lower().startswith('pm_')}
+
+    remove = pm_remove | {c.lower() for c in EBI_REMOVE}
+    to_drop = [c for c in df.columns if c.lower() in remove]
+
+    return df.drop(columns=to_drop, inplace=False)
```

Comment:
I'm confused.  Is df.drop not case sensitive?  We set all of the columns to drop to be lowercase, but I don't remember that all of the columns of the dataframe were lowercase to begin with.

Reply:
it is. this is probably out of an abundance of caution

Reply:
If it is case sensitive... then won't this fail to drop any columns that have capital letters in them in the input dataframe?  Capital letters seem to be expected and allowed based on the regex in in the construction of multiselect answers.

Reply:
The `to_drop` variable is based on unaltered entries in `df.columns` -- the case of the elements in `to_drop` has not been altered.

Reply:
(the list comprehension on line 40)